pkg/authorize/kubecertmanager/authorize.go

```diff
@@ -98,42 +128,139 @@ func (c *authorizer) validateIngress(cx *api.Context, ingress *extensions.Ingres
 		return append(errs, field.Invalid(field.NewPath("annotations").Key(label), value,
 			fmt.Sprintf("invalid kube-cert-manager provider type: %s, expected: %s", value, class)))
 	}
-
+	
 	// @check the nginx is external
 	label = "kubernetes.io/ingress.class"
 	class = "nginx-external"
-
+	
 	if value, found := annontations[label]; !found {
 		return append(errs, field.Invalid(field.NewPath("annotations").Key(label), "",
 			"the ingress does not specify a nginx class in annotations"))
 	} else if value != class {
 		return append(errs, field.Invalid(field.NewPath("annotations").Key(label), value,
 			fmt.Sprintf("invalid kube-cert-manager provider, expected '%s' for a http challenge", class)))
 	}
+	
+	errs = append(errs, c.validateIngressPointed(cx, ingress)...)
+	
+	return errs
+}
 
-	// @step: get namespace for this object
-	namespace, err := utils.GetCachedNamespace(cx.Client, cx.Cache, ingress.Namespace)
-	if err != nil {
-		return append(errs, field.InternalError(field.NewPath("namespace"), err))
+func (c *authorizer) validateCertManagerIngress(cx *api.Context, ingress *extensions.Ingress) field.ErrorList {
+	var errs field.ErrorList
+	
+	if managers := getCertManagerReferences(ingress); len(managers) !=1 || managers[0] != "cert-manager.io" {
+		return nil
 	}
-
-	enableDNSCheck := true
-	if v, found := namespace.GetAnnotations()[cx.Annotation(EnableDNSCheck)]; found {
-		if b, err := strconv.ParseBool(v); err == nil {
-			enableDNSCheck = b
+	
+	// there is at least one annotation or label prefixed with "cert-manager.io"
+	if value, found := ingress.GetAnnotations()["cert-manager.io/enabled"]; !found || value != "true" {
+	  	return append(errs, field.Invalid(field.NewPath("metadata.annotations.cert-manager.io/enabled"), "", "cert-manager.io annotations or labels are present, but annotation cert-manager.io/enabled: \"true\" is missing"))
+    }
+	
+    ingressValueValid := false
+	ingressValue, ingressFound := ingress.GetAnnotations()["kubernetes.io/ingress.class"]
+	if ingressFound {
+		solverValue, solverFound := ingress.GetLabels()["cert-manager.io/solver"]
+		if ingressValue == "nginx-internal" {
+			ingressValueValid = true
+		    if !solverFound || solverValue != "route53" {
+				errs = append(errs, field.Invalid(field.NewPath("metadata.labels.cert-manager.io/solver"), solverValue, "nginx-internal has been specified as an annotation for kubernetes.io/ingress.class but label cert-manager.io/solver is missing or not set to route53"))
+			}
+			
+			errs = append(errs, c.isHostedInternally(ingress)...)
+			errs = append(errs, c.validateIngressPointed(cx, ingress)...)
+		} else if ingressValue == "nginx-external" {
+			ingressValueValid = true
+			if solverFound && solverValue != "http01" {
```

Comment:
Is this check valid? Why do we want to prevent using the route53 resolver if it is an external ingress resource? Both would work.

Reply:
You're right. I remember trying it in the past and it not working, but maybe I didn't wait long enough before giving up on it.
I've retested it with nginx-external and route53 solver and it works. It took a couple of minutes to get a valid cert but it did work.
So will amend the code and modify the unit tests